src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -27,160 +72,895 @@ const ActorInit Obj_Bean_InitVars = {
+    Math_StepToF(&this->unk_1CC, 0.0f, 0.1f);
+    Math_StepToF(&this->unk_1D0, 0.0f, 0.1f);
+    Math_ScaledStepToS(&this->dyna.actor.shape.rot.y, this->dyna.actor.home.rot.y, 100);
+}
+
+void func_80937238(ObjBean* this) {
+    this->dyna.actor.world.pos.x = this->unk_1BC.x + this->unk_1CC;
+    this->dyna.actor.world.pos.y = this->unk_1BC.y;
+    this->dyna.actor.world.pos.z = this->unk_1BC.z + this->unk_1D0;
+}
+
+void func_80937268(ObjBean* this, GlobalContext* globalCtx) {
+    this->unk_1D8 = globalCtx->setupPathList[OBJBEAN_GET_3F00(&this->dyna.actor)].count - 1;
+    this->unk_1DA = 0;
+    this->unk_1DC = 1;
+}
+
+void func_809372A8(ObjBean* this) {
+    Math_Vec3s_ToVec3f(&this->unk_1BC, this->unk_1D4);
+}
+
+void func_809372D0(ObjBean* this) {
+    Actor* actor = &this->dyna.actor;
+    Vec3f sp38;
+    f32 sp34;
+    f32 temp_f2;
+    f32 temp_f12;
+
+    Math_Vec3s_ToVec3f(&sp38, &this->unk_1D4[this->unk_1DC]);
+    Math_Vec3f_Diff(&sp38, &this->unk_1BC, &actor->velocity);
+
+    sp34 = Math3D_Vec3fMagnitude(&actor->velocity);
+    temp_f2 = D_80938FF8[this->unk_1DE].x;
+    temp_f12 = D_80938FF8[this->unk_1DE].y;
+    if (sp34 < (actor->speedXZ * 8.0f)) {
+        temp_f2 = ((temp_f2 - 2.0f) * 0.1f) + 2.0f;
+        temp_f12 *= 0.4f;
+    }
+
+    Math_StepToF(&actor->speedXZ, temp_f2, temp_f12);
+
+    if ((actor->speedXZ + 0.05f) < sp34) {
+        Math_Vec3f_Scale(&actor->velocity, actor->speedXZ / sp34);
+        this->unk_1BC.x += actor->velocity.x;
+        this->unk_1BC.y += actor->velocity.y;
+        this->unk_1BC.z += actor->velocity.z;
+    } else {
+        Math_Vec3f_Copy(&this->unk_1BC, &sp38);
+        this->unk_1DA = this->unk_1DC;
+        if (this->unk_1DA >= this->unk_1D8) {
+            this->unk_1DC = 0;
+        } else {
+            this->unk_1DC++;
+        }
+        actor->speedXZ *= 0.5f;
+    }
+}
+
+s32 func_80937468(ObjBean* this, GlobalContext* globalCtx) {
+    Actor* bgActor = globalCtx->actorCtx.actorList[ACTORCAT_BG].first;
+
+    while (bgActor != NULL) {
+        if ((bgActor->id == ACTOR_EN_HORSE) &&
+            (Math3D_Vec3fDistSq(&bgActor->world.pos, &this->dyna.actor.world.pos) < 10000.0f)) {
+            return true;
+        }
+        bgActor = bgActor->next;
+    }
+
+    return false;
+}
+
+Actor* func_809374F8(ObjBean* this, GlobalContext* globalCtx) {
+    Actor* bgActor = globalCtx->actorCtx.actorList[ACTORCAT_BG].first;
+    s32 params = OBJBEAN_GET_3F80(&this->dyna.actor, 0);
+
+    while (bgActor != NULL) {
+        if (bgActor->id == ACTOR_OBJ_BEAN) {
+            s32 params2 = OBJBEAN_GET_C000(bgActor);
+
+            if (!params2 && (bgActor->room == this->dyna.actor.room) && !OBJBEAN_GET_80(bgActor) &&
+                (params == OBJBEAN_GET_7F(bgActor, 0)) &&
+                (Math3D_Vec3fDistSq(&bgActor->world.pos, &this->dyna.actor.world.pos) < 100.0f)) {
+                break;
+            }
+        }
+        bgActor = bgActor->next;
+    }
+
+    return bgActor;
+}
```

Comment:
From what I can see, this can only return an `ObjBean*` or `NULL`. Any reason to not just return `ObjBean*`? Would also prevent needing to cast in `func_809375C8`.